packages/core/src/pages/api/preview.ts

```diff
@@ -22,17 +22,38 @@ class StatusError extends Error {
 
 const pickParam = (req: NextApiRequest, parameter: string) => {
   const maybeParam = req.query[parameter]
+  return typeof maybeParam === 'string' ? maybeParam : undefined
+}
+
+const getSeoSlug = (page: any): string | undefined => {
+  if (contentSource.type === 'CP') {
```

Comment:
This is a minor change, but how about support both `CP` or `cp`? Just to make it harder for users to run into issues due to typos. 

(if it's safe to do so)
```suggestion
  if (contentSource.type.toLocaleLowerCase() === 'CP') {
```

Reply:
I made a new small function to handle this validation https://github.com/vtex/faststore/pull/2807/commits/b58e0bdb904380ddafec2fd0251dca79e65f1ef0#diff-3ddf02e0a1d0dfb1b0a37c8b8aec07fcca2614ef2d15abb46cae4e1c3ce2104cR4